servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -109,20 +110,83 @@ private static List<String> getHeaderValues(final HttpHeaders headers, final Str
 
     private static final class ClientGetter extends ServiceTalkHttpAttributesGetter
             implements HttpClientAttributesGetter<HttpRequestMetaData, HttpResponseMetaData> {
+
+        @Override
         @Nullable
+        public String getUrlFull(final HttpRequestMetaData request) {
+            HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
+            if (effectiveHostAndPort == null) {
+                return null;
+            }
+            StringBuilder sb = new StringBuilder();
+            String scheme = request.scheme();
+            sb.append(scheme == null ? "http" : scheme)
+              .append("://").append(effectiveHostAndPort.hostName());
+            if (effectiveHostAndPort.port() >= 0) {
+                sb.append(':').append(effectiveHostAndPort.port());
+            }
+            sb.append(request.requestTarget());
```

Comment:
Is this a required attribute? It's not easy to construct it for the following reasons:
1. `effectiveHostAndPort` is always `null` for single-address client because `HostHeaderHttpRequesterFilter` is added later in the pipeline.
2. `scheme` is also `null` bcz the request factory doesn't know if a client has TLS or not
3. Including full `requestTarget` is a high risk of leaking sensitive information, like user identity or secret hashes

Can we start with always returning `null` here?

Reply:
Ok, I see that it was already there. Then let's keep it for now, but we need to reconsider how to make it safer in the future.

Also, pre-existing code used concatenation instead of a `StringBuilder` to avoid resizing `StringBuilder` internally. Consider keeping the code as-is or use estimated length for `StringBuilder` constructor.

Reply:
Concatenation is not necessarily cheaper, especially when creating intermediates. The JVM is free to decide, but it often gets turned into a StringBuilder process.

Reply:
I estimated the length. Using the StringBuilder makes it a bit easier to omit the port if it's implicit based on the scheme.